server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -380,17 +379,15 @@ public boolean syncUpTo(long offset) throws IOException {
                         closeWithTragicEvent(ex);
                         throw ex;
                     }
+                    flushedSequenceNumbers.forEach((LongProcedure) persistedSequenceNumberConsumer::accept);
```

Comment:
I tried to keep this out of the syncLock mutex, but it's trickier than I thought. The reason is that the other translog-sync can drain flushedSequenceNumbers then we have no operation to mark as persisted before returning.

Reply:
Since we go through the AsyncIOProcessor for all indexing fsync's, I think this is not a problem.